Decode DuckDB blobs as buffers in Node UDF args

matt-allan opened the pull request:
When the UDF arguments include a blob, pass the argument to Node as a buffer instead of a string to ensure the bytes are preserved.

I ran into this while trying to write a few user defined functions to decode binary data. I couldn't work around the issue because of the conversion to UTF-16 that appears to happen when the string is passed to Node.

Commit:
Decode DuckDB blobs as buffers in Node UDF args

When the UDF arguments include a blob, pass the argument to Node
as a buffer instead of a string to ensure the bytes are preserved.

Mytherin commented on tools/nodejs/src/data_chunk.cpp:
```diff
@@ -145,7 +145,20 @@ Napi::Array EncodeDataChunk(Napi::Env env, duckdb::DataChunk &chunk, bool with_t
+
+					for (size_t i = 0; i < chunk.size(); ++i) {
+						auto value = data[i].GetString();
+						auto buf = Napi::Buffer<char>::Copy(env, value.c_str(), value.length());
```
Calling `.GetString()` performs a copy, effectively leading to two copies being performed here. Can we instead use `data[i].GetDataUnsafe()` and `data[i].GetSize()`?

matt-allan replied:
Thanks, just pushed a fix!

Mytherin reviewed (commented):
Thanks for the PR! LGTM. One comment:

Commit:
Avoid unnecessary second copy of blob to Node buffer

The blob is already copied by the Napi Buffer so we don't need to
call `GetString` on the DuckDB string to copy it beforehand.

Mytherin commented on tools/nodejs/src/data_chunk.cpp:
```diff
@@ -145,7 +145,19 @@ Napi::Array EncodeDataChunk(Napi::Env env, duckdb::DataChunk &chunk, bool with_t
 			}
-			case duckdb::LogicalTypeId::BLOB:
+			case duckdb::LogicalTypeId::BLOB: {
+				if (with_data) {
```
Completely unrelated to this PR - but `with_data` and `with_types` on this method appear to me like legacy parameters. `EncodeDataChunk` does not appear to be called anywhere where these settings are not both set to true. @Mause is that correct or am I missing something?

Mytherin reviewed (commented):
Thanks for the fixes! LGTM. Perhaps @Mause also wants to have a look?

matt-allan commented:
Thank you @Mytherin, just wanted to say I've had a really positive experience interacting with the DuckDB community so far :heart:
